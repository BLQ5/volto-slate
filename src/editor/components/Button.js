import cx from 'classnames';
import React from 'react';
import { Icon } from '@plone/volto/components';

// import { Button as UIButton } from 'semantic-ui-react';

export const Button = React.forwardRef(
  ({ className, active, reversed, icon, style, ...props }, ref) => {
    style = {
      ...style,
      cursor: 'pointer',
      color: reversed
        ? active
          ? 'white'
          : '#888'
        : active
        ? ' black'
        : '#888',
    };
    return (
      <span {...props} ref={ref} style={style} className={cx(className)}>
        <Icon name={icon} size="24px" />
      </span>
    );
  },
);
