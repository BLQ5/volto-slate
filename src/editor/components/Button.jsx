import cx from 'classnames';
import React from 'react';
import { Icon } from '@plone/volto/components';

// import { Button as UIButton } from 'semantic-ui-react';

const Button = React.forwardRef(
  ({ className, active, reversed, icon, style, ...props }, ref) => {
    style = {
      ...style,
      cursor: 'pointer',
      padding: '4px !important',
      width: '32px',
      height: '32px',
      display: 'inline-block',
      color: reversed
        ? active
          ? 'white'
          : '#888'
        : active
        ? ' black'
        : '#888',
    };
    return (
      <span {...props} ref={ref} style={style} className={cx(className)}>
        {icon ? <Icon name={icon} size="24px" /> : 'no-icon'}
      </span>
    );
  },
);

export default Button;
