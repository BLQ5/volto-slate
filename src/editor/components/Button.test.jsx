import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import { wait } from '@testing-library/react';

import Button from './Button';

const mockStore = configureStore();

describe('Button', () => {
  it('renders a button', async () => {
    const store = mockStore({
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <Button pathname="/test" />
      </Provider>,
    );
    await wait(() => {
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
