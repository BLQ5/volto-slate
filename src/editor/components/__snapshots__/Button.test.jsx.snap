// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a button 1`] = `
<span
  className=""
  pathname="/test"
  style={
    Object {
      "color": "#888",
      "cursor": "pointer",
    }
  }
>
  no-icon
</span>
`;
